toolkit/tools/liveinstaller/liveinstaller.go

```diff
@@ -119,19 +125,69 @@ func installerFactory(forceAttended bool, configFile, templateConfigFile string)
 	}
 
 	if isAttended {
-		installFunc = func(args imagerArguments) (bool, error) {
+		installFunc = func(args imagerArguments) (installationDetails, error) {
 			return terminalUIAttendedInstall(templateConfigFile, args)
 		}
 	} else {
-		installFunc = func(args imagerArguments) (bool, error) {
+		installFunc = func(args imagerArguments) (installationDetails, error) {
 			return unattendedInstall(configFile, args)
 		}
 	}
 
 	return
 }
 
-func updateBootOrder() (err error) {
+func updateBootOrder(installDetails installationDetails) (err error) {
+	if (installDetails.finalConfig.DefaultSystemConfig.BootType != "efi") {
+		logger.Log.Info("No BootType of 'efi' detected. Not attempting to set EFI boot order.")
+		return
+	}
+
+	err = removeOldMarinerBootTargets()
+	if (err != nil) {
+		return
+	}
+
+	logger.Log.Info("Setting Boot Order")
+	const squashErrors = false
```

Comment:
nit: I think we generally put const variables at the top of the function (unless you have seen this elsewhere, then ignore)

Reply:
Will change make this change! Thank you!